File: qqbot-runtime/src/main/java/com/mieai/qqbot/runtime/security/StaticKeyProvider.java
package com.mieai.qqbot.runtime.security;

import java.util.Objects;
import java.util.Optional;
import javax.crypto.spec.SecretKeySpec;

/** Single-master-key provider suitable for application configuration and local deployments. */
public final class StaticKeyProvider implements KeyProvider {
    private final Optional<KeyMaterial> keyMaterial;

    private StaticKeyProvider(Optional<KeyMaterial> keyMaterial) {
        this.keyMaterial = keyMaterial;
    }

    public static StaticKeyProvider configured(String keyId, byte[] keyBytes) {
        Objects.requireNonNull(keyBytes, "keyBytes must not be null");
        return new StaticKeyProvider(Optional.of(
                new KeyMaterial(keyId, new SecretKeySpec(keyBytes, "AES"))));
    }

    public static StaticKeyProvider unconfigured() {
        return new StaticKeyProvider(Optional.empty());
    }

    public boolean isConfigured() {
        return keyMaterial.isPresent();
    }

    @Override
    public KeyMaterial activeKey() {
        return keyMaterial.orElseThrow(KeyUnavailableException::new);
    }

    @Override
    public Optional<KeyMaterial> findById(String keyId) {
        Objects.requireNonNull(keyId, "keyId must not be null");
        if (keyMaterial.isEmpty()) {
            throw new KeyUnavailableException();
        }
        return keyMaterial.filter(material -> material.keyId().equals(keyId));
    }

    @Override
    public String toString() {
        return "StaticKeyProvider[configured=" + keyMaterial.isPresent() + ", key=<redacted>]";
    }
}

File: qqbot-runtime/src/main/kotlin/com/mieai/qqbot/runtime/security/StaticKeyProvider.kt
package com.mieai.qqbot.runtime.security

import javax.crypto.spec.SecretKeySpec

class StaticKeyProvider private constructor(private val keyMaterial: KeyMaterial?) : KeyProvider {
    val isConfigured: Boolean
        get() = keyMaterial != null

    override fun activeKey(): KeyMaterial = keyMaterial ?: throw KeyUnavailableException()

    override fun findById(keyId: String): KeyMaterial? {
        val configured = keyMaterial ?: throw KeyUnavailableException()
        return configured.takeIf { it.keyId == keyId }
    }

    override fun toString(): String =
        "StaticKeyProvider[configured=${keyMaterial != null}, key=<redacted>]"

    companion object {
        fun configured(keyId: String, keyBytes: ByteArray) =
            StaticKeyProvider(KeyMaterial(keyId, SecretKeySpec(keyBytes, "AES")))

        fun unconfigured() = StaticKeyProvider(null)
    }
}
